### ISSUE TITLE
[Translations][Prezent] Doctrine mapping and component dependencies issue

### ISSUE THREAD

**Issue by koemeet** [Sylius/Sylius#2268]
Hi,

I was rebasing my fork to the latest of Sylius and I noticed some very big flaws. The biggest one is that the mapping of the prezent translations (`<prezent:translatable>`) are hardcoded and cannot be overwritten. This breaks all applications that are overriding the default models.

The second problem is that the `Attribtue` component for example has a hard dependency on the `Product` component. See this line in the file `AttributeTranslation.orm.xml`:

``` xml
<prezent:translatable field="translatable" target-entity="Sylius\Component\Product\Model\Attribute" locale="locale"/>
```

I think this issue is already known by the core team, but wanted to make an issue for it so we can discuss on how to solve this.

@pjedrzejewski @gonzalovilaseca Are you guys planning on solving this?

**Comment by inspiran** [comment]
It is not a perfect fix, but I have this:

   protected function getTranslationEntityClass()
    {
        $class = get_class($this).'Translation';

        if (!class_exists($class)) {
            $class = get_parent_class($this).'Translation';
        }

        return $class;
    }

**Comment by gonzalovilaseca** [comment]
@steffenbrem This should be fixed by #2314
